Tribler/Main/vwxGUI/home.py

```diff
@@ -1059,8 +1059,14 @@ def SetData(self, delayedResult):
         for torrent in torrents:
             data.append((torrent.infohash, [torrent.name], torrent, ThumbnailListItemNoTorrent))
 
-        self.list.SetData(data)
-        self.list.SetupScrolling()
+        # This may throw a PyDeadObject error when a test is finishing
+        # but a delayedtask (GuiDBHandler) calls SetData with delayedResult.get().
```

Comment:
I do not know exactly where this task is being scheduled. I tried to add the self and self.list check after the delayedtask.get function but that didn't help either.

I noticed that this workaround was implemented a couple of times throughout tribler, so I decided to go with this.

If you'd like I can try to find out where it is being scheduled, but at this point the only thing I know is that the SetData function is being scheduled to be called.